x-pack/solutions/security/plugins/security_solution/public/sourcerer/components/index.tsx

```diff
@@ -291,56 +284,43 @@ export const Sourcerer = React.memo<SourcererComponentProps>(({ scope: scopeId }
     sourcererMissingPatterns,
   ]);
 
-  // deprecated timeline index pattern handlers
-  const onContinueUpdateDeprecated = useCallback(() => {
-    setIsShowingUpdateModal(false);
-    const patterns = selectedPatterns.filter((pattern) =>
-      defaultDataView.patternList.includes(pattern)
-    );
-    dispatchChangeDataView(defaultDataView.id, patterns);
-    setPopoverIsOpen(false);
-  }, [defaultDataView.id, defaultDataView.patternList, dispatchChangeDataView, selectedPatterns]);
-
-  const onUpdateDeprecated = useCallback(() => {
-    // are all the patterns in the default?
-    if (missingPatterns.length === 0) {
-      onContinueUpdateDeprecated();
-    } else {
-      // open modal
-      setIsShowingUpdateModal(true);
-    }
-  }, [missingPatterns, onContinueUpdateDeprecated]);
-
-  const [isTriggerDisabled, setIsTriggerDisabled] = useState(false);
-
-  const onOpenAndReset = useCallback(() => {
+  const handleDataViewFallbackError = useCallback(() => {
     setPopoverIsOpen(true);
     resetDataSources();
   }, [resetDataSources]);
 
-  const updateDataView = useUpdateDataView(onOpenAndReset);
-  const onUpdateDataView = useCallback(async () => {
-    const isUiSettingsSuccess = await updateDataView(missingPatterns);
-    setIsShowingUpdateModal(false);
-    setPopoverIsOpen(false);
+  const handleApplyFallbackDataView = useCallback(
+    (dataView: DataViewSpec) => {
+      if (!dataView.id) {
+        return;
+      }
 
-    if (isUiSettingsSuccess) {
-      dispatchChangeDataView(
-        defaultDataView.id,
-        // to be at this stage, activePatterns is defined, the ?? selectedPatterns is to make TS happy
-        activePatterns ?? selectedPatterns,
-        false
-      );
-      setIsTriggerDisabled(true);
+      dispatchChangeDataView(dataView.id, dataView.title?.split(',') || [], false, dataView);
+      setDataViewId(dataView.id);
```

Comment:
I know this is additional work, but I think we should be pulling the `dataViewId` from sourcerer state rather than relying on some mix of local and redux state here 🤔. I wonder if we're seeing any bugs because of this...